mesh/mesh.cpp

```diff
@@ -3595,11 +3595,20 @@ Mesh::Mesh(Mesh *orig_mesh, int ref_factor, int ref_type)
       }
    }
 
-   SetCurvature(1, true, spaceDim);
-   Vector node_coordinates_vec(
-      node_coordinates.Data(),
-      node_coordinates.Width()*node_coordinates.Height());
-   SetNodes(node_coordinates_vec);
+   if (orig_mesh->GetNodes())
+   {
+      Vector node_coordinates_vec(
+         node_coordinates.Data(),
+         node_coordinates.Width()*node_coordinates.Height());
+      L2_FECollection fec_dg(1, Dim, BasisType::GaussLobatto);
+      FiniteElementSpace fes_dg(this, &fec_dg, spaceDim, 1);
+      GridFunction nodes_dg(&fes_dg);
+      nodes_dg = node_coordinates_vec;
+
+      bool discont = orig_mesh->GetNodalFESpace()->IsDGSpace();
+      SetCurvature(1, discont, spaceDim);
```

Comment:
Since this is a virtual call in a constructor, I'm not sure if this will call the parallel version when constructing a `ParMesh` (the `ParMesh` constructor calls this constructor). If this calls the parallel `SetCurvature` then there is no need to call `SetCurvature` again from the parallel constructor.

Reply:
I also wasn't sure what would happen here, but it turns out that the call to `SetCurvature` in the `Mesh` constructor won't call `ParMesh::SetCurvature`, which is the cause of the issue in #1361.

Reply:
I think I've run into this somewhat unexpected behavior before. However, this seems to be the right behavior according to the section "During construction and destruction" in https://en.cppreference.com/w/cpp/language/virtual.